### DISCUSSION TITLE
How to enable hashing

### DISCUSSION THREAD

**Discussion by Tosheen** [unocss/unocss#3032] [Q&A] [answered]
I would like to enable hashing of the class names, so for example  class name of `bg-primary-200` should get converted to something like uno-ytzh12s or anything similar, doesn't really matter.

I am aware of the transformer but that's not what I would like to use.  I do not want to create a single class which is combination of multiple properties. 

Is there an easy way to do so?

**Comment by Shyam-Chen** [comment]
### `@apply`: 

```scss
.button {
  @apply flex justify-center items-center gap-2 px-6 py-2 border rounded uppercase text-sm font-medium;
  @apply shadow-md hover:shadow-lg;
  @apply focus:outline-none focus:ring-2 focus:ring-primary-500/40;
  
  &.primary {
    @apply bg-primary-500 text-white border-primary-500;
    @apply hover:text-blueGray-300 hover:bg-primary-600;
    @apply active:bg-primary-400 dark:active:bg-primary-800;
  }
}
```

**Reply by Tosheen** [reply]
Thank you but its not really what I am looking for even though I get the idea behind this.

**Comment by zyyv** [comment, marked as answer]
`transformer-compile-class` is currently not able to compile hash for a single unitilty, pr welcome to enhance its behavior. Or I think you can try to write a new transformer. Its principle is to put the compiled unitilty with hash into `shortcuts`.

**Reply by Tosheen** [reply]
Ok so its not really possible at the moment.
That was on of the features of the twind that I really liked.

Hopefully it will be available in the future.

**Reply by JenuelDev** [reply]
https://github.com/unocss/unocss/tree/main/packages/transformer-compile-class

**Comment by JenuelDev** [comment]
Any Update on this?

**Comment by beebeefish** [comment]
<img width="1352" alt="image" src="https://github.com/user-attachments/assets/3662b266-b02d-4047-9a4f-a619d1fec60a">
[Pin](https://www.pinterest.com/)Do you know Pinterest's method for hashing individual applications

**Comment by zyyv** [comment]
FYI, I made a simple transformer, which currently only supports hashing of classes, and does not support attribute mode. Welcome to PR for enhancement.

https://github.com/unpreset/unocss-transformer-hash

![image](https://github.com/user-attachments/assets/09297584-aef6-4eab-a16d-c2cb76c8e07e)
